### PR TITLE
added label entity

### PR DESCRIPTION
**Pull request by VlasenkoYevhenii**

### CODE REVIEW

File: src/main/java/app/model/Label.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private String color;
+
+    @ManyToMany
```
Comment (makaroshyna) [inline src/main/java/app/model/Label.java:31]: Do you have to have this, or does Task entity should have a list of labels?

File: src/main/java/app/model/Label.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.Table;
+import java.util.List;
+import lombok.Data;
+import org.hibernate.annotations.SoftDelete;
+
+@Entity
+@Table(name = "labels")
+@SoftDelete(columnName = "is_deleted")
+@Data
```
Comment (makaroshyna) [inline src/main/java/app/model/Label.java:19]: Maybe better to use only setter and getter annotations? It could cause problems with task IDs.

**Review by makaroshyna — CHANGES_REQUESTED** [review]

File: src/main/java/app/model/Label.java (hunk 1 above)
Comment (VlasenkoYevhenii) [reply src/main/java/app/model/Label.java:31]: resolved by swapping join column and reversejoincolumn

File: src/main/java/app/model/Label.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private Long id;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private String color;
+
+    @ManyToMany(mappedBy = "labels")
+    private List<Task> taskIds;
```
Comment (makaroshyna) [inline src/main/java/app/model/Label.java:32]: Delete this, I don't think we need task IDs here at all.

**Review by makaroshyna — CHANGES_REQUESTED** [review]

**Review by makaroshyna — APPROVED** [review]

**Review by ValentynSharshon — APPROVED** [review]

**Pull request merged**